### PR TITLE
chore(format): fix double quotes and alter the location of refs and names in relations

### PR DESCRIPTION
**Pull request by cpcloud**
@kszucs Here's an example of the new repr:

```
UnboundTable[r0, name=purchases]
  region string
  kind   string
  user   int64
  amount float64

Aggregation[r1]
  table:
    ref: r0
  metrics:
    total: float64 = Sum
      amount: float64 = r0.amount
      where:
        None
  by:
    region: string = r0.region
    kind: string = r0.kind
  predicates:
    boolean = Equals
      left:
        kind: string = r0.kind
      right:
        value: string = 'foo'

Aggregation[r2]
  table:
    ref: r0
  metrics:
    total: float64 = Sum
      amount: float64 = r0.amount
      where:
        None
  by:
    region: string = r0.region
    kind: string = r0.kind
  predicates:
    boolean = Equals
      left:
        kind: string = r0.kind
      right:
        value: string = 'bar'

Selection[r3]
  table:
    InnerJoin
      left:
        ref: r1
      right:
        ref: r2
      predicates:
        boolean = Equals
          left:
            region: string = r1.region
          right:
            region: string = r2.region
  selections:
    ref: r1
    right_total: float64 = r2.total
```

### CODE REVIEW

File: ibis/tests/expr/test_format.py [hunk 1]
```diff
@@ -206,16 +207,26 @@ def test_argument_repr_shows_name():
...
 def test_same_column_multiple_aliases():
     table = ibis.table([('col', 'int64')], name='t')
     expr = table[table.col.name('fakealias1'), table.col.name('fakealias2')]
     result = repr(expr)
 
-    assert 'fakealias1' in result
-    assert 'fakealias2' in result
+    assert "UnboundTable[r0, name=t]" in result
+    assert "col int64" in result
+    assert "fakealias1 = r0.col :: int64" in result
```
Comment (kszucs) [inline ibis/tests/expr/test_format.py:223]: Based on the example in the PR's description the double colon shouldn't be present anymore.

Comment (cpcloud) [reply ibis/tests/expr/test_format.py:223]: Yep, let me fix up the tests.

Comment (kszucs) [reply ibis/tests/expr/test_format.py:223]: I'd prefer to have a simple example like in the description asserted in test_format.py

File: ibis/expr/format.py [hunk 2]
```diff
@@ -113,33 +113,33 @@ def get_result(self):
...
-            text = f'ScalarParameter[{self._get_type_display()}]'
+            text = self._get_type_display()
         elif isinstance(what, ops.Node):
             text = self._format_node(self.expr)
 
         if isinstance(self.expr, ir.ValueExpr) and self.expr._name is not None:
-            text = f'{self.expr.get_name()} = {text}'
+            raw_name = self.expr.get_name()
+            if isinstance(self.expr.op(), ops.ScalarParameter):
+                name = f"$({raw_name})"
```
Comment (kszucs) [inline ibis/expr/format.py:125]: Nice!

File: ibis/tests/expr/test_format.py [hunk 3]
```diff
@@ -206,16 +207,43 @@ def test_argument_repr_shows_name():
...
+    # XXX: This is the only exact repr test. Do
+    # not add new exact repr tests. New repr tests
+    # should check for the presence of substrings.
+    table = ibis.table(
+        [('col', 'int64'), ("col2", "string"), ("col3", "double")],
+        name='t',
+    )
+    result = repr(table)
+    expected = """\
+UnboundTable[r0, name=t]
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/tests/expr/test_format.py:245]: I can add a `Selection` here. I moved all the repr tests to this file, so it should be more convenient to change/update the repr as desired.

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/tests/expr/test_format.py:245]: Done.

**Review by saulpw — APPROVED** [review]
Looks good to me!

**Comment by cpcloud** [comment]
@jreback Mind approving?

**Review by jreback — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
completely OT @cpcloud how did you get the 'update branch' button to appear?

**Comment by cpcloud** [comment]
Thanks, will merge once the CI is passing on #3319.

**Comment by cpcloud** [comment]
> completely OT @cpcloud how did you get the 'update branch' button to appear? e.g. where is this setting

@jreback I didn't do anything special for that, not sure!

**Comment by jreback** [comment]
ok thanks!

**Pull request merged**